karibu-dsl-v10/build.gradle.kts

```diff
@@ -8,7 +8,7 @@ dependencies {
 
     // Vaadin
     compile("com.vaadin:vaadin-core:${ext["vaadin10_version"]}")
-    compile("javax.servlet:javax.servlet-api:3.1.0")
+    compileOnly("javax.servlet:javax.servlet-api:3.1.0")
```

Comment:
`compileOnly` is supposed to work exactly like `provided` with Maven: https://maven.apache.org/guides/introduction/introduction-to-dependency-mechanism.html#Dependency_Scope
The library such as Karibu-DSL doesn't know what kind of environment it will run in, and therefore it can't declare `compileOnly` dependency - it must use the `compile` dependency which can then be overridden later on, for example by the WAR project, or the Spring Boot App project.

The typical way for all libraries is to declare `compile` deps; such deps are then overridden as compileOnly when the app is bundled as a WAR file or a Spring Boot uber-jar. The Spring Boot uber-jar can then either enforce the servlet-api version, or exclude servlet-api from karibu-dsl or use other Maven- or Gradle- dependent tricks.

That being said, it is true that servlet api is somewhat special. It is clear that Karibu-DSL will always run in a servlet container, with two exceptions:
* Karibu-Testing
* some special VertX/Netty setup, but we can safely ignore this
Since Karibu-Testing brings Servlet API in, we can also ignore that use-case.

Therefore it's clear that Karibu-DSL will always run in an environment where Servlet API is provided. This way of thinking is apparently also employed by Vaadin itself: the vaadin-server.jar also links to servlet API via `provided`.

Reply:
With the above, I agree to make Servlet API `provided`.